lib/src/rules/use_late_for_private_fields_and_variables.dart

```diff
@@ -163,10 +167,12 @@ class _Visitor extends UnifyingAstVisitor<void> {
     if (variable.isSynthetic) {
       return;
     }
-    if (context.typeSystem.isNonNullable(variable.declaredElement!.type)) {
+    var declaredElement = variable.declaredElement;
+    if (declaredElement == null ||
+        context.typeSystem.isNonNullable(declaredElement.type)) {
       return;
     }
-    lateables[currentUnit]!.add(variable);
+    lateables[currentUnit]?.add(variable);
```

Comment:
No change required, just some musings: This one feels a little odd, like we're saying "if we're keeping data about this key, then add the new data". On the other hand the only alternative I've thought of (`lateables.putIfAbsent(currentUnit, () => ...).add(variable)`) doesn't seem great either because we don't expect to ever need to invoke the closure.

Reply:
100% agreed.  @a14n: I think this is making a mess of your code a bit.  Maybe after the dust settles you could take a pass?  (Or I could.)  Sorry for that.